### PR TITLE
JDK-8315278: Patch 'print-targets' target to print targets separated by new line

### PR DESCRIPTION
**Pull request by JohnTortugo**
Currently 'make print-targets' print all targets on the same line separated by space. That isn't only terribly to read but hard to "grep". Printing each target on a separate line seems much better.

Tested on: Linux x64, Windows x64 and MacOS AArch64

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315278](https://bugs.openjdk.org/browse/JDK-8315278): Patch 'print-targets' target to print targets separated by new line (**Enhancement** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Koichi Sakata](https://openjdk.org/census#ksakata) (@jyukutyo - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15479/head:pull/15479` \
`$ git checkout pull/15479`

Update a local copy of the PR: \
`$ git checkout pull/15479` \
`$ git pull https://git.openjdk.org/jdk.git pull/15479/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15479`

View PR using the GUI difftool: \
`$ git pr show -t 15479`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15479.diff">https://git.openjdk.org/jdk/pull/15479.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15479#issuecomment-1698426630)

### CODE REVIEW

**Comment by erikj79** [comment]
I'm not sure about this change. There are three similar and somewhat related diagnostic targets `print-targets` `print-modules` and `print-tests` that currently behave in the same way. If we are to change that behavior, then I think it needs to be uniform. What is your usecase for `print-targets`? Are you using it to find appropriate targets using grep manually? If so, I can see why newline separation would be preferable. I just want to be sure that we aren't breaking some other common usecase.

Historically, `print-targets` was first introduced to help support bash completion by having a way of communicating the full target list from Main.gmk to Init.gmk. That was changed in JDK-8076465 where we started caching the list of targets in a generated makefile to improve performance, and `print-targets` was left as a diagnostics tool.

File: make/Main.gmk [hunk 1]
```diff
@@ -1425,7 +1425,7 @@ ALL_TARGETS += $(addsuffix -only, $(filter-out dist-clean clean%, $(ALL_TARGETS)
 # are internal only, to support Init.gmk.
 
 print-targets:
-	  @$(ECHO) $(sort $(ALL_TARGETS))
+	  @$(ECHO) $(sort $(ALL_TARGETS)) | $(TR) ' ' '\n'
```
Comment (erikj79) [inline make/Main.gmk:1428]: If we are to do this, I would suggest reducing the amount of external commands used and keeping it to make functions, and also fixing the indentation.
```suggestion
	$(info $(subst $(SPACE),$(NEWLINE),$(sort $(ALL_TARGETS))))
```
I would also suggest applying the same change to all three print targets.

**Comment by JohnTortugo** [comment]
Hi, @erikj79 . Thank you for reviewing.

> What is your usecase for print-targets? Are you using it to find appropriate targets using grep manually?

Yes, that's exactly how I often use print-targets and I end up always having to append a `tr ' ' '\n'`. I thought that others may have the same issue so I decided to create this patch.

**Review by erikj79 — APPROVED** [review]
After having investigated the history of these targets and verified that they aren't used for anything internal to the makefiles that might break, I think this is ok and will improve usability.

**Comment by JohnTortugo** [comment]
Thank you @erikj79

**Review by jyukutyo — APPROVED** [review]

**Comment by jyukutyo** [comment]
I'll sponsor you.

/sponsor

**Pull request closed without merging**